File: app/src/main/java/com/dewonderstruck/apps/ashx0/db/AboutUsDao.java
package com.dewonderstruck.apps.ashx0.db;

import com.dewonderstruck.apps.ashx0.viewobject.AboutUs;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

/**
 * Created by Vamsi Madduluri on 12/30/17.
 * Contact Email : vamsii.wrkhost@gmail.com
 */

@Dao
public interface AboutUsDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(AboutUs aboutUs);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<AboutUs> aboutUsList);

    @Query("SELECT * FROM AboutUs LIMIT '1'")
    LiveData<AboutUs> get();

    @Query("SELECT * FROM AboutUs")
    LiveData<List<AboutUs>> getAll();

    @Query("DELETE FROM AboutUs")
    void deleteTable();

}

File: app/src/main/java/com/dewonderstruck/apps/ashx0/db/AboutUsDao.kt
package com.dewonderstruck.apps.ashx0.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.dewonderstruck.apps.ashx0.viewobject.AboutUs

/**
 * Created by Vamsi Madduluri on 12/30/17.
 * Contact Email : vamsii.wrkhost@gmail.com
 */
@Dao
interface AboutUsDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(aboutUs: AboutUs?)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(aboutUsList: List<AboutUs?>?)

    @Query("SELECT * FROM AboutUs LIMIT '1'")
    fun get(): LiveData<AboutUs?>?

    @get:Query("SELECT * FROM AboutUs")
    val all: LiveData<List<AboutUs?>?>?

    @Query("DELETE FROM AboutUs")
    fun deleteTable()
}